TYP: Inconsistent behavior when assigning arrays of a scalar type to arrays using the scalar's alias

### Describe the issue:

There exists an inconsistency in numpy 2.3 when assigning an array typed to have a scalar type such as `np.double` to an array variable typed to have its matching alias such as `np.float64` as dtype: For all types except `np.float64` and `np.complex128`, the assignment passes type checking with mypy, but fails for the other two. (The example below is limited to the floating point numbers, but similar things happen for complex floating numbers.)

This also happens when replacing the scalar type, e.g. when replacing `np.double` with `np.floating[np._typing._nbit_base._64Bit]`, again with only assignment to `float64` failing and all others passing.

From reading some of the recent issues, it was my impression that the assignment of `np.floating` to any more concrete dtype should be forbidden, but this seems to only be the case for the aforementioned two types. In any case, I would expect consistent behavior across all three examples. Am I missing a crucial detail here? Or is this an artifact of the changes to `np.float64` and `np.complex128` introduced in recent versions?

### Reproduce the code example:

```python
import numpy as np
import numpy.typing as npt


x_1: npt.NDArray[np.double]= np.array([1], dtype=np.double)
reveal_type(x_1)
y_1: npt.NDArray[np.float64] = x_1

x_2: npt.NDArray[np.half] = np.array([1], dtype=np.half)
reveal_type(x_2)
y_2: npt.NDArray[np.float16] = x_2

x_3: npt.NDArray[np.single] = np.array([1], dtype=np.single)
reveal_type(x_3)
y_3: npt.NDArray[np.float32] = x_3
```

### Error message:

```shell
shape2.py:6: note: Revealed type is "numpy.ndarray[builtins.tuple[Any, ...], numpy.dtype[numpy.floating[numpy._typing._nbit_base._64Bit]]]"
shape2.py:7: error: Incompatible types in assignment (expression has type "ndarray[tuple[Any, ...], dtype[floating[_64Bit]]]", variable has type "ndarray[tuple[Any, ...], dtype[float64]]")  [assignment]
shape2.py:10: note: Revealed type is "numpy.ndarray[builtins.tuple[Any, ...], numpy.dtype[numpy.floating[numpy._typing._nbit_base._16Bit]]]"
shape2.py:14: note: Revealed type is "numpy.ndarray[builtins.tuple[Any, ...], numpy.dtype[numpy.floating[numpy._typing._nbit_base._32Bit]]]"
Found 1 error in 1 file (checked 1 source file)
```

### Python and NumPy Versions:

2.3.0
3.12.11 | packaged by conda-forge | (main, Jun  4 2025, 14:29:09) [MSC v.1943 64 bit (AMD64)]

### Type-checker version and settings:

mypy 1.16.0

Command used: `mypy --strict shape2.py`

### Additional typing packages.

mypy_extensions 1.1.0, typing_extensions 4.14.0, typing-inspection 0.4.1, annotated_types 0.7.0

No stub packages.

--- a/numpy/__init__.pyi
+++ b/numpy/__init__.pyi
@@ -4215,9 +4215,9 @@ class float64(floating[_64Bit], float):  # type: ignore[misc]
     def __divmod__(self, other: _Float64_co, /) -> _2Tuple[float64]: ...  # type: ignore[override]
     def __rdivmod__(self, other: _Float64_co, /) -> _2Tuple[float64]: ...  # type: ignore[override]
 
-half: TypeAlias = floating[_NBitHalf]
-single: TypeAlias = floating[_NBitSingle]
-double: TypeAlias = floating[_NBitDouble]
+half: TypeAlias = float16
+single: TypeAlias = float32
+double: TypeAlias = float64
 longdouble: TypeAlias = floating[_NBitLongDouble]
 
 # The main reason for `complexfloating` having two typevars is cosmetic.
@@ -4315,7 +4315,7 @@ class complexfloating(inexact[_NBit1, complex], Generic[_NBit1, _NBit2]):
         self, other: number[_NBit], mod: None = None, /
     ) -> complexfloating[_NBit1, _NBit2] | complexfloating[_NBit, _NBit]: ...
 
-complex64: TypeAlias = complexfloating[_32Bit, _32Bit]
+complex64: TypeAlias = complexfloating[_32Bit]
 
 class complex128(complexfloating[_64Bit, _64Bit], complex):  # type: ignore[misc]
     @overload
@@ -4376,9 +4376,9 @@ class complex128(complexfloating[_64Bit, _64Bit], complex):  # type: ignore[misc
     ) -> complexfloating[_NBit1 | _64Bit, _NBit2 | _64Bit]: ...
     def __rpow__(self, other: _Complex128_co, mod: None = None, /) -> complex128: ...
 
-csingle: TypeAlias = complexfloating[_NBitSingle, _NBitSingle]
-cdouble: TypeAlias = complexfloating[_NBitDouble, _NBitDouble]
-clongdouble: TypeAlias = complexfloating[_NBitLongDouble, _NBitLongDouble]
+csingle: TypeAlias = complex64
+cdouble: TypeAlias = complex128
+clongdouble: TypeAlias = complexfloating[_NBitLongDouble]
 
 class timedelta64(_IntegralMixin, generic[_TD64ItemT_co], Generic[_TD64ItemT_co]):
     @property

